src/main/scala/pathy/path.scala

```diff
@@ -178,6 +180,11 @@ object Path {
   def parentDir[B,T,S](path: Path[B,T,S]): Option[Path[B,Dir,S]] =
     peel(path).map(_._1)
 
+  def fileParent[B,S](file: Path[B, File, S]): Path[B, Dir, S] = file match {
```

Comment:
In this case, the fact that the path type is `File` guarantees it must also have a parent dir by construction (i.e. there is no way, other than `asInstanceOf`, to create an instance of `Path[_,File,_]` that isn't the `FileIn` constructor) so I'm comfortable without one.